lib/cylc/taskdef.py

```diff
@@ -163,9 +161,9 @@ def get_cleanup_cutoff_point(cls, my_point, offset_sequence_tuples):
                 cutoff_points.append(my_cutoff_point)
         if cutoff_points:
             max_cutoff_point = max(cutoff_points)
-            if max_cutoff_point < my_point:
-                # This is caused by future triggers - default to my_point.
-                return my_point
+            if max_cutoff_point < point:
+                # This is caused by future triggers - default to point.
+                return point
             return max_cutoff_point
-        # There aren't any dependent tasks in other cycles for my_point.
-        return my_point
+        # There aren't any dependent tasks in other cycles for point.
+        return point
```

Comment:
I was originally looking at this method for clues of what went wrong, hence the minor refactor. There is actually nothing wrong with this method, apart from the rather awkward interface, which is simplified somewhat in this change.